### ISSUE TITLE
IndexError: Replacement index 2 out of range for positional args tuple

### ISSUE THREAD

**Issue by vladimirtiukhtin** [pytorch/serve#2501]
### 🐛 Describe the bug

I just switched from pytorch/torchserve-kfs:0.8.0-gpu to pytorch/torchserve-kfs:0.8.1-gpu changing nothing else and it broke my kserve installation. Rolling it back to 0.8.0 helped

### Error logs
```
2023-07-27 09:33:43.941 uvicorn.access INFO:     192.0.2.1:0 14 - "POST /v1/models/test%3Apredict HTTP/1.1" 500 Internal Server Error
2023-07-27 09:33:43.941 uvicorn.error ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/venv/lib/python3.9/site-packages/uvicorn/protocols/http/h11_impl.py", line 373, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/home/venv/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
  File "/home/venv/lib/python3.9/site-packages/fastapi/applications.py", line 270, in __call__
    await super().__call__(scope, receive, send)
  File "/home/venv/lib/python3.9/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/venv/lib/python3.9/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/venv/lib/python3.9/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/venv/lib/python3.9/site-packages/timing_asgi/middleware.py", line 68, in __call__
    await self.app(scope, receive, send_wrapper)
  File "/home/venv/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/home/venv/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/home/venv/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/venv/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/venv/lib/python3.9/site-packages/starlette/routing.py", line 706, in __call__
    await route.handle(scope, receive, send)
  File "/home/venv/lib/python3.9/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/home/venv/lib/python3.9/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/home/venv/lib/python3.9/site-packages/fastapi/routing.py", line 235, in app
    raw_response = await run_endpoint_function(
  File "/home/venv/lib/python3.9/site-packages/fastapi/routing.py", line 161, in run_endpoint_function
    return await dependant.call(**values)
  File "/home/venv/lib/python3.9/site-packages/kserve/protocol/rest/v1_endpoints.py", line 69, in predict
    response, response_headers = await self.dataplane.infer(model_name=model_name, body=body, headers=headers)
  File "/home/venv/lib/python3.9/site-packages/kserve/protocol/dataplane.py", line 276, in infer
    response = await model(body, headers=headers)
  File "/home/venv/lib/python3.9/site-packages/kserve/model.py", line 117, in __call__
    response = (await self.predict(payload, headers)) if inspect.iscoroutinefunction(self.predict) \
  File "/home/venv/lib/python3.9/site-packages/kserve/model.py", line 320, in predict
    return await self._http_predict(payload, headers)
  File "/home/venv/lib/python3.9/site-packages/kserve/model.py", line 256, in _http_predict
    predict_url = PREDICTOR_URL_FORMAT.format(self.predictor_host, self.name)
IndexError: Replacement index 2 out of range for positional args tuple
2023-07-27 09:33:47.402 uvicorn.access INFO:     10.102.5.111:0 14 - "GET /metrics HTTP/1.1" 200 OK
```
### Installation instructions

Kserve 0.10.0 with manually raised torchserve version via InferenceService manifest
```
resource "kubernetes_manifest" "test_model" {
  manifest = {
    apiVersion = "serving.kserve.io/v1beta1"
    kind       = "InferenceService"
    metadata = {
      name      = "test"
      namespace = kubernetes_namespace_v1.test.metadata.0.name
      annotations = {
        "prometheus.io/scrape" = "true"
        "prometheus.io/port"   = "8082"
      }
    }
    spec = {
      predictor = {
        serviceAccountName = kubernetes_service_account_v1.kserve_model.metadata.0.name
        minReplicas        = 1
        maxReplicas        = 3
        timeout            = 60
        batcher = {
          maxBatchSize = 10
          maxLatency   = 80
        }
        model = {
          name : "test"
          image = "pytorch/torchserve-kfs:0.8.1"
          modelFormat = {
            name = "pytorch"
          }
          resources = {
            requests = {
              "cpu" : "2"
              "memory" : "16Gi"
              "nvidia.com/gpu" : "1"
            }
            limits = {
              "cpu" : "4"
              "memory" : "24Gi"
              "nvidia.com/gpu" : "1"
            }
          }
          storageUri = "s3://${module.ml_models_bucket.bucket_name}/test"
        }
      }
    }
  }
}
```

**Comment by agunapal** [comment]
Hi @jagadeeshi2i  , tagging you in case you are aware of this

**Comment by jagadeeshi2i** [comment]
The issue is on Kserve v0.10.0 and is fixed on v0.11.0. By rebuilding the `pytorch/torchserve-kfs` docker image with Kserve v0.11.0 the issue can be resolved.

**Comment by jagadeeshi2i** [comment]
Fixed with latest `0.8.2` release. Refer `pytorch/torchserve-kfs:0.8.2` image.

**Comment by jagadeeshi2i** [comment]
Fixed with latest `0.8.2` release. Refer `pytorch/torchserve-kfs:0.8.2` image.
